services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java

```diff
@@ -513,12 +520,46 @@ public void onTick(Date date) {
     on_tick(transaction, currentTime);
     transaction.commit().join();
 
-    if (nextSlotDue) {
-      if (syncService.isSyncActive()) {
-        processSlotWhileSyncing();
-      } else {
-        processSlot();
-      }
+    if (nextSlotDue && syncService.isSyncActive()) {
+      processSlotWhileSyncing();
+      return;
+    }
+
+    final UnsignedLong genesisTime = recentChainData.getGenesisTime();
+    final UnsignedLong calculatedSlot =
+        currentTime.compareTo(genesisTime) < 0
+            ? ZERO
+            : ForkChoiceUtil.getCurrentSlot(currentTime, genesisTime);
+    if (currentTime.compareTo(genesisTime) < 0) {
+      return;
+    }
```

Comment:
We bail out if we're prior to geneesis time so can just do that check before the calculatedSlot calculation.

Reply:
fair point.